ui/ducks/send/send.js

```diff
@@ -0,0 +1,1238 @@
+          state.amount.error = NEGATIVE_ETH_ERROR;
+          break;
+        // If none of the above are true, set error to null
+        default:
+          state.amount.error = null;
+      }
+    },
+    validateGasField: (state) => {
+      // Checks if the user has enough funds to cover the cost of gas, always
+      // uses the native currency and does not take into account the amount
+      // being sent. If the user has enough to cover cost of gas but not gas
+      // + amount then the error will be displayed on the amount field.
+      const insufficientFunds = !isBalanceSufficient({
+        amount: '0x0',
+        balance: state.account.balance,
+        gasTotal: state.gas.gasTotal ?? '0x0',
+      });
+
+      state.gas.error = insufficientFunds ? INSUFFICIENT_FUNDS_ERROR : null;
+    },
+    validateSendState: (state) => {
+      switch (true) {
+        // 1 + 2. State is invalid when either gas or amount fields have errors
+        // 3. State is invalid if asset type is a token and the token details
+        //  are unknown.
+        // 4. State is invalid if no recipient has been added
+        // 5. State is invalid if gasLimit is less than the minimumGasLimit
+        case Boolean(state.amount.error):
+        case Boolean(state.gas.error):
+        case state.asset.type === ASSET_TYPES.TOKEN &&
+          state.asset.details === null:
+        case state.stage === SEND_STAGES.ADD_RECIPIENT:
+        case new BigNumber(state.gas.gasLimit, 16).lessThan(
+          new BigNumber(state.gas.minimumGasLimit),
+        ):
+          state.status = SEND_STATUSES.INVALID;
+          break;
+        default:
+          state.status = SEND_STATUSES.VALID;
+      }
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(QR_CODE_DETECTED, (state, action) => {
+        // When data is received from the QR Code Scanner we set the recipient
+        // as long as a valid address can be pulled from the data. If an
+        // address is pulled but it is invalid, we display an error.
+        const qrCodeData = action.value;
+        if (qrCodeData) {
+          if (qrCodeData.type === 'address') {
+            const scannedAddress = qrCodeData.values.address.toLowerCase();
+            if (isValidAddress(scannedAddress)) {
+              if (state.recipient.address !== scannedAddress) {
+                slice.caseReducers.updateRecipient(state, {
+                  payload: { address: scannedAddress },
+                });
+              }
+            } else {
+              state.recipient.error = INVALID_RECIPIENT_ADDRESS_ERROR;
+            }
+          }
+        }
+      })
+      .addCase(SELECTED_ACCOUNT_CHANGED, (state, action) => {
+        // This event occurs when the user selects a new account from the
+        // account menu, or the currently active account's balance updates.
+        state.account.balance = action.payload.account.balance;
+        state.account.address = action.payload.account.address;
```

Comment:
needs to check if see if we are in edit flow!

Reply:
Implemented a new event for accounts that are not the selected account, for when on the edit flow. This event handler has been changed to preclude the edit flow.